xbmc/storage/MediaManager.h

```diff
@@ -8,6 +8,7 @@
 
 #pragma once
 
+#include "FileItem.h"
```

Comment:
Regarding this, I think it may be better to forward declare the CFileItem in the header, and do the include in MediaManager.cpp.

Probably explains why it worked for windows, FileItem.h has been included in some other header that's behind a windows ifdef

Reply:
that should all work ok.
would by interesting to see if it still builds on windows with leaving the forward declaration behind.

Reply:
Still best practice to have the include in the implementation file imo. Don't rely on an unknown include from somewhere else, as if that other header is changed you get a random breakage. CFileItem is absolutely used in the implementation, so declare it.
I believe that very scenario has affected some Devs recently when trying to do some minor change that snowballed all because of the luck of an include being done in some other include file.

Reply:
yeah, that's for sure. no need to check the win-build without forward-decl. unless @yeahme49 wants to ;) just confirm that all suggested include-removals do work

Reply:
moved the include and forward declared CFileItem in the header